src/lib/RuleResultRecombinator.ts

```diff
@@ -78,16 +79,19 @@ export class RuleResultRecombinator {
 		for (const result of results) {
 			const from = process.cwd();
 			const fileName = path.relative(from, result.fileName);
+			const escapedFileName = this.safeHtmlEscape(fileName);
```

Comment:
We should consider importing an xml package that will do this for us. Added extra escaping found in tests as a short term fix.